src/EditorFeatures/Core/DefinitionContextTracker.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.SymbolMapping;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor
+{
+    [Export(typeof(IWpfTextViewConnectionListener))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Interactive)]
+    internal class DefinitionContextConnectionListener : IWpfTextViewConnectionListener
+    {
+        private readonly Dictionary<ITextView, CancellationTokenSource> _views = new Dictionary<ITextView, CancellationTokenSource>();
```

Comment:
renamed to `_subscribedViews` and converted to a `HashSet<ITextView>` after taking @jasonmalinowski's other suggestion of using a single task instead of one per view.